main.tf

```diff
@@ -64,15 +64,26 @@ resource "google_compute_instance" "main" {
       nat_ip = google_compute_address.static.address
     }
   }
+
+  metadata_startup_script = <<EOT
+curl -L https://tljh.jupyter.org/bootstrap.py | sudo python3 - --admin ${var.admin-username}
+sudo tljh-config set https.enabled true
+sudo tljh-config set https.letsencrypt.email ${var.letsencrypt-email}
+sudo tljh-config add-item https.letsencrypt.domains ${var.subdomain}.${var.domain}
+sudo tljh-config reload proxy
+sudo tljh-config set user_environment.default_app jupyterlab
+sudo tljh-config reload hub
+sudo -E conda install -c conda-forge numpy pandas scipy
```

Comment:
Considering that we can install requirements from a file, it would be best to pin them in an `environment.yml` file to ensure at least some level of reproducibility there see https://tljh.jupyter.org/en/latest/topic/customizing-installer.html#installing-python-packages-in-the-user-environment

Reply:
Would you be okay with this being out of scope? A change like this would require integrating something like ansible (which I do agree in order to to this properly would be the right approach). Basically the user script only runs once on deployment and will not update the cluster on changes.

Reply:
No that is not true. And if that were the only option we can revert to just install everything with pip and pass the file URL directly here as in the docs. There is no need for extra tooling.
I can sort that on my end.

Reply:
> No that is not true. And if that were the only option we can revert to just install everything with pip and pass the file URL directly here as in the docs

I think the manual solution is the easiest. Of running `sudo -E conda install -c conda-forge numpy pandas scipy` within jupyterlab. 

But to have this automated in infrastructure we would either need ansible or something with terraform that runs every time. The `user-config` section is oneshot and only applies during the creation of the resource.

Reply:
See https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#metadata_startup_script